pkg/checks/external/main.go

```diff
@@ -124,14 +127,29 @@ func init() {
 }
 
 // New creates a new external checker
-func New(client *kubernetes.Clientset, checkConfig *khcheckcrd.KuberhealthyCheck, khCheckClient *khcheckcrd.KuberhealthyCheckClient, khStateClient *khstatecrd.KuberhealthyStateClient, reportingURL string) *Checker {
+func New(client *kubernetes.Clientset, checkConfig *khcheckcrd.KuberhealthyCheck, jobConfig *khjobcrd.KuberhealthyJob, khCheckClient *khcheckcrd.KuberhealthyCheckClient, khStateClient *khstatecrd.KuberhealthyStateClient, khJobClient *khjobcrd.KhjobV1Client, reportingURL string, khjob bool) *Checker {
```

Comment:
Consider refactor to `NewChecker` and `NewJob` funcs.   This lets us remove the `khjob bool` from this function signature.  New checks will continue to call `New` and `New` will simply call `NewChecker`.  Likewise, `NewJob` will create a new job.